ecosystem/sep-0040.md

```diff
@@ -0,0 +1,160 @@
+
+In order to minimize the costs of on-chain storage, we have multiple levels of indirection of NFT metadata. The only thing the NFT account stores on-chain is a description of where to find the metadata.
+
+#### NFT Account Description
+A proper NFT account should specify its `homedomain`. It should also configure the asset in its domain's `stellar.toml` file (see [SEP-1](https://stellar.org/sep-1#currency-documentation)) if appropriate.
+
+There's only one required field on the account itself:
+
+| Field       | Purpose |
+| ----------- | ------- |
+| `type`      | Describes how the "first hop" of metadata is defined, standard values include `ipfshash` and `stellar` |
+
+The two possible values for the `type` field describe two standard ways to store the metadata, and the details will be in a data entry corresponding to this value. In other words, `type: ipfshash` means that you should look under the `ipfshash` key data entry which further defines how to find the metadata.
+
+All other values for the `type` are considered non-standard until they are approved for inclusion within this SEP.
+
+Fields should be stored as simple key-value pairs within an account's data entries. The standard limitations of `ManageDataOp` apply: field names and their values can only be 64 bytes each. Most metadata attributes should be stored at the NFT reference location itself.
+
+**Backward Compatibility:** Existing NFT platforms essentially default to `type: ipfshash`. While this standard aims to free up the ecosystem from dependence on IPFS, it should also work with existing conventions that developed organically. To do so, **if you see an account with a lone `ipfshash` data entry, you may treat it as an NFT account with `type: ipfshash`**.
+
+##### IPFS Storage
+If the `type` is `ipfshash`, the `ipfshash` key must point to an [IPFS Content IDentifier](https://docs.ipfs.io/concepts/content-addressing/) corresponding to the location of the metadata.
+
+If you are the issuer, make sure you understand the implications of using IPFS. You must use an IPFS lookup service to actually resolve to the metadata. For example, [this URL](https://cloudflare-ipfs.com/ipfs/QmaydGEshbL7HMQkD6FZPcMeEt6o15aC4p4P82VxQ7amjy) resolves to the Meridian 2021 Hat NFT via the Cloudflare IPFS service. If you're the issuer, you must also ensure that IPFS actually stores your data (by running your own node or entrusting a provider).
+
+##### Stellar Storage
+If the `type` is `stellar`, the `stellar` key must point to a Stellar account ID. That account will store the metadata within its data entries. Refer to the unofficial [SEP-39](https://github.com/stellar/stellar-protocol/pull/1090) specification for details on how to read an account's data entries for the NFT metadata.
```

Comment:
There isn't anything such as an official or unofficial SEP in my opinion, if that SEP is SEP-39 we should merge it, or if not drop this section until SEP-39 is merged.
```suggestion
If the `type` is `stellar`, the `stellar` key must point to a Stellar account ID. That account will store the metadata within its data entries. Refer to [SEP-39](SEP-0039.md) specification for details on how to read an account's data entries for the NFT metadata.
```